.happy/terraform/modules/schema_migration/main.tf

```diff
@@ -224,11 +224,15 @@ resource aws_sfn_state_machine sfn_schema_migration {
       },
       "Next": "SpanCollections",
       "Retry": [ {
-          "ErrorEquals": ["AWS.Batch.TooManyRequestsException", "Batch.BatchException", "Batch.AWSBatchException"],
+          "ErrorEquals": ["States.TaskFailed"],
+          "IntervalSeconds": 30,
+          "MaxAttempts": 1
+      }, {
+          "ErrorEquals": ["States.ALL"],
```

Comment:
The only error this will catch is `State.Timeout` according to the AWS docs. Is that what is intended?

Reply:
Where are you seeing that this would only catch `States.Timeout`? I do see from the [AWS docs](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html) this note:
```
The States.ALL error type can't catch the States.DataLimitExceeded terminal error type and runtime error types. For more information about these error types, see [States.DataLimitExceeded](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html#error-data-limit-exceed) and [States.Runtime](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html#states-runtime-error).
```

Regardless, I think your suggestion to explicitly catch the AWS batch errors we were previously catching is a good one, so I'll update to do that.

Reply:
Because `States.TaskFailed` would catch every thing else and except for `States.Timeout`.

From [AWS docs](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html)
>States.TaskFailed
A Task state failed during the execution. When used in a retry or catch, States.TaskFailed acts as a wildcard that matches any known error name except for States.Timeout.